Update to Django 1.7

Hwesta opened the pull request:
Update Archivematica to use Django 1.7.  Several changes made to update backwards incompatible changes, notably that ModelForms must specify their fields, and that standalone scripts (all the clientScripts) must setup Django before calling it.

Hwesta committed:
Templates: Remove load url from future

refs #7305

Not needed in Django 1.5, deprecated in 1.7

Hwesta committed:
ATK: Replace django's simplejson with json

refs #7305

Deprecated in Django 1.7, and builtin json module is equivalent.

Hwesta committed:
Dashboard: Update mimetype to content_type in HttpResponse

refs #7305

mimetype is deprecated in Django 1.7, replaced by content_type

Hwesta committed:
ATK, AS: Update Form to take advantage of ModelForm

Use ModelForm's ability to autogenerate fields from the Model, instead
of redefining fields. Move labels and choices to the model.

Hwesta committed:
Forms: Add fields listing to all ModelForms

refs #7305

Django 1.7 no longer defaults to fields being everything because that
is a security risk. Add explicit specification of fields to all
ModelForms.

Hwesta committed:
Update WSGI, manage.py for Djange 1.7

refs #7305

Hwesta committed:
Models: Set defaults on BooleanFields

refs #7305

Hwesta committed:
Django: Update to 1.7

refs #7305

Update Django and Tastypie versions. Update test runner default to
suppress warnings. Update all client scripts to initialize Django
before using the ORM.

mistydemeo commented:
:+1:

Hwesta commented:
In progress updating for #98 changes

mistydemeo commented on src/MCPClient/lib/clientScripts/archivematicaTranscribeFile.py:
```diff
@@ -8,18 +8,14 @@
-if path not in sys.path:
-    sys.path.append(path)
+import django
+sys.path.append('/usr/share/archivematica/dashboard')
```
I wonder if it's worth changing MCPClient to export `PYTHONPATH` and `DJANGO_SETTINGS_MODULE` when spinning up subprocesses? That way we could avoid requiring every client script to set its own `sys.path`.

mistydemeo replied:
(Not that that needs to be in this PR, it just occurred to me.)

Hwesta replied:
That is a great idea, looking into it.  We use threads not subprocesses currently, so I dunno how different it'll be...

mistydemeo replied:
We still run them as separate CLI scripts, so they're definitely being spun off as new processes at some level. IIRC our subprocess wrapper does support exporting the environment too.

Hwesta replied:
This was fixed in #161 (after I forgot this conversation and did the work anyway)

mistydemeo commented on src/dashboard/src/components/rights/forms.py:
```diff
@@ -81,12 +79,14 @@ class Meta:
 class RightsCopyrightNoteForm(ModelForm):
     class Meta:
         model = models.RightsStatementCopyrightNote
+        fields = ('copyrightnote',)  # TODO check this is correct
```
Yep, `copyrightNote` is the only non-pk/non-fk field

mistydemeo commented on src/dashboard/src/external/archivematica-fpr-admin:
```diff
@@ -1 +1 @@
-Subproject commit c253d9e447344c3369c581025acaf016d8e114d4
+Subproject commit e87f9196d70eef1f5c1525c851859dd8bcd66568
```
Was this intentional?

Hwesta replied:
Yes - updates to match artefactual/archivematica-fpr-admin#8

mistydemeo commented:
Still :+1:

Hwesta committed:
Dashboard: Misc updates reducing warnings

get_query_set deprecated in favor of get_queryset. cycle now takes
variables and literals must be quoted.

Hwesta commented:
Includes #161

mistydemeo commented:
Still :+1:

Hwesta committed:
Postinst: Upgrade pip dependencies, handle pyo's

refs #7305

Pip does not remove pyo's that were generated after install, so remove
those before upgrading Django as it causes a dirty upgrade. Change pip
command to update dependencies.

Hwesta committed:
URLs: Update to new URL syntax

refs #7305

All urls created with the url() function.  Patterns will be going away
in 1.8, so standardize how it's imported and used to make removing it
easier later.

Hwesta committed:
Common tests: Fix test

The DB enforces that events must point to a valid file UUID.

Hwesta commented:
Added updates to postinst so the package upgrade should work, but needs to be tested.  Updated one of the databaseFunction tests, because updating to 1.7 broke it somehow. Responded to feedback.

mistydemeo commented:
:shipit:
